src/common/libsdprocess/sdprocess.c

```diff
@@ -0,0 +1,1830 @@
+{
+    struct flux_sdprocess *sdp = NULL;
+    flux_reactor_t *reactor;
+
+    if (!h || !unitname) {
+        errno = EINVAL;
+        return NULL;
+    }
+
+    if (!(reactor = flux_get_reactor (h)))
+        return NULL;
+
+    if (!(sdp = sdprocess_create (h,
+                                  reactor,
+                                  unitname,
+                                  NULL,
+                                  NULL,
+                                  -1,
+                                  -1,
+                                  -1)))
+        return NULL;
+
+    if (check_exist (sdp) < 0)
+        goto cleanup;
+
+    return sdp;
+
+cleanup:
+    flux_sdprocess_destroy (sdp);
+    return NULL;
+}
+
+flux_sdprocess_t *flux_sdprocess_local_find_unit (flux_reactor_t *r,
+                                                  const char *unitname)
+{
+    struct flux_sdprocess *sdp = NULL;
+
+    if (!r || !unitname) {
+        errno = EINVAL;
+        return NULL;
+    }
+
+    if (!(sdp = sdprocess_create (NULL,
+                                  r,
+                                  unitname,
+                                  NULL,
+                                  NULL,
+                                  -1,
+                                  -1,
+                                  -1)))
+        return NULL;
+
+    if (check_exist (sdp) < 0)
+        goto cleanup;
+
+    return sdp;
+
+cleanup:
+    flux_sdprocess_destroy (sdp);
+    return NULL;
+}
+
+static int get_final_properties (flux_sdprocess_t *sdp)
+{
+    sd_bus_error error = SD_BUS_ERROR_NULL;
+    uint32_t exec_main_status;
+    char *result = NULL;
+    int save_errno, ret, rv = -1;
+
+    if ((ret = sd_bus_get_property_trivial (sdp->bus,
+                                            "org.freedesktop.systemd1",
+                                            sdp->service_path,
+                                            "org.freedesktop.systemd1.Service",
+                                            "ExecMainStatus",
+                                            &error,
+                                            'i',
+                                            &exec_main_status)) < 0) {
+        set_errno_log_error (sdp->h, ret, &error, "sd_bus_get_property_trivial");
+        goto cleanup;
+    }
+
+    if ((ret = sd_bus_get_property_string (sdp->bus,
+                                           "org.freedesktop.systemd1",
+                                           sdp->service_path,
+                                           "org.freedesktop.systemd1.Service",
+                                           "Result",
+                                           &error,
+                                           &result)) < 0) {
+        set_errno_log_error (sdp->h, ret, &error, "sd_bus_get_property_string");
+        goto cleanup;
+    }
+    
+    sdp->exec_main_status = exec_main_status;
+    sdp->result = result;
```

Comment:
Valgrind caught a leak here. Do you need to `free (sdp->result)` before setting the new value?